### PR TITLE
Fix simple cppcheck issues

### PR DESCRIPTION
**Pull request by matthiakl**
Re #1406
I worked through approx. half of the sources fixing simple cppcheck issues (mostly `const` and `explicit` declarations).
I also updated the `create_cppcheck_report` to work with newest version and github links.

[cppcheck_report.html](https://github.com/widelands/widelands/files/5367268/cppcheck_report.html.txt)

### CODE REVIEW

File: src/editor/map_generator.cc [hunk 1]
```diff
@@ -846,7 +846,7 @@ int UniqueRandomMapInfo::map_id_char_to_number(char ch) {
 	} else if ((ch == '1') || (ch == 'l') || (ch == 'L') || (ch == 'I') || (ch == 'i') ||
 	           (ch == 'J') || (ch == 'j')) {
 		return 23;
-	} else if (ch >= 'A' && ch < 'O') {
+	} else if (ch >= 'A' && ch < 'Z') {
```
Comment (gunchleoc) [inline src/editor/map_generator.cc]:
```suggestion
	} else if (ch >= 'A' && ch <= 'Z') {
```

File: src/logic/map_objects/tribes/soldier.cc [hunk 2]
```diff
@@ -1231,9 +1231,6 @@ void Soldier::defense_update(Game& game, State& state) {
 	if (soldiers.empty() || (get_current_health() < get_retreat_health())) {
 		if (get_retreat_health() > get_current_health()) {
 			assert(state.ivar1 & CF_RETREAT_WHEN_INJURED);
-		}
-
-		if (get_current_health() < get_retreat_health()) {
```
Comment (gunchleoc) [inline src/logic/map_objects/tribes/soldier.cc:1236]: Swap the molog calls below - the comparator is the other way around

**Comment by gunchleoc** [comment]
Testing done, so when the 2 comments are fixed, this can go in.

File: src/logic/map_objects/tribes/soldier.cc (hunk 2 above)
Comment (matthiakl) [reply src/logic/map_objects/tribes/soldier.cc:1236]: They are the same: the comparator is the other way around, but the operands are also switched. Both conditions match the one in line 1231, so I changed it to this notation for easier understanding. The `else` part corresponds to `soldiers.empty()`.
Thanks for the fast review :)

**Review by gunchleoc — APPROVED** [review]

**Comment by gunchleoc** [comment]
We have a test failure now as well as a bunch of new compiler warnings https://travis-ci.org/github/widelands/widelands/jobs/735362661

**Comment by matthiakl** [comment]
The compiler warning should be fixed, however I can't reproduce it locally (clang-10).
Testcase is also fixed. I added a comment to prevent future cppcheck reviews from the same problem.
+1 for the testsuite beating cppcheck :)

**Comment by gunchleoc** [comment]
There's a merge conflict now

**Pull request merged**